$OpenBSD: patch-xwindow_c,v 1.7 2015/11/09 14:36:12 dcoppa Exp $

commit c9ff826fbef2b9d4df84c1ef5ec5eba33661abee
Author: Uli Schlachter <psychon@znc.in>
Date:   Mon Apr 14 10:04:14 2014 +0200

xwindow_shape_pixmap: Correctly handle invalid geometry

Signed-off-by: Uli Schlachter <psychon@znc.in>

commit 4313d93c6c5b09565ed331c3e9a7653069d5337d
Author: Uli Schlachter <psychon@znc.in>
Date:   Sat Oct 10 17:38:51 2015 +0200

Fix obvious typo in xwindow_translate_for_gravity()

Signed-off-by: Uli Schlachter <psychon@znc.in>

--- xwindow.c.orig	Sat Jan 10 20:36:35 2015
+++ xwindow.c	Mon Nov  9 13:54:50 2015
@@ -330,6 +330,9 @@ xwindow_shape_pixmap(int width, int height, cairo_surf
     cairo_surface_t *dest;
     cairo_t *cr;
 
+    if (width <= 0 || height <= 0)
+        return XCB_NONE;
+
     xcb_create_pixmap(globalconf.connection, 1, pixmap, globalconf.screen->root, width, height);
     dest = cairo_xcb_surface_create_for_bitmap(globalconf.connection, globalconf.screen, pixmap, width, height);
 
@@ -413,7 +416,7 @@ void xwindow_translate_for_gravity(xcb_gravity_t gravi
         break;
     case XCB_GRAVITY_STATIC:
         x = -change_width_before;
-        x = -change_height_before;
+        y = -change_height_before;
         break;
     }
 
